src/ObjectUtils/ElfFile.cpp

```diff
@@ -120,6 +124,15 @@ class ElfFileImpl : public ElfFile {
   std::vector<orbit_grpc_protos::ModuleInfo::ObjectSegment> loadable_segments_;
 };
 
+bool IsHotpatchable(const absl::flat_hash_set<uint64_t>& hotpachable_addresses,
+                    uint64_t symbol_address) {
+  // The hotpatchable addresses stored in the elf file point to the first byte of the padding. We
+  // require the binary to be compiled with a five byte padding and a two byte nop at the function
+  // entry. So we check for "address - 5" to be listed as hotpatchable.
```

Comment:
Maybe I'm misunderstanding something here: Does this mean that the symbol start address is after the padding? But the function entry (where calls land) has to be with the nops, right (o/w hot patching wouldn't actually work)? What does this mean for samples that fall into the padding / nops when we do symbol resolution?

Reply:
Clarified in person, I indeed misunderstoodd. The function entry is after the padding, but then there are still the nops at the function entry. So, this is fine for symbol resolution.